src/main/java/com/epam/architecture/SOAPws/impl/UserSOAPServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.epam.architecture.SOAPws.impl;
+
+import com.epam.architecture.SOAPws.UserSOAPService;
+import com.epam.architecture.entities.Book;
+import com.epam.architecture.userinterface.LibraryService;
+import com.sun.xml.ws.server.AbstractWebServiceContext;
+import jakarta.annotation.PreDestroy;
+import jakarta.annotation.Resource;
+import jakarta.jws.HandlerChain;
+import jakarta.jws.WebService;
+import jakarta.xml.soap.*;
+import jakarta.xml.ws.WebServiceContext;
+import jakarta.xml.ws.soap.SOAPFaultException;
+
+import java.util.Iterator;
+
+@HandlerChain(file = "../user-handler.xml")
+@WebService(endpointInterface = "com.epam.architecture.SOAPws.UserSOAPService")
+public class UserSOAPServiceImpl implements UserSOAPService {
+
+    @Resource
+    private WebServiceContext context;
+
+    private LibraryService libraryService = LibraryService.getInstanceWithDeserializeData();
+
+    @Override
+    public boolean addBookmark(String isbn, int pageNumber) throws SOAPException {
+        return libraryService.addBookmark(getLogin(), isbn, pageNumber);
+    }
+
+    @Override
+    public boolean deleteBookmark(String isbn, int pageNumber) throws SOAPException {
+        return libraryService.deleteBookmark(getLogin(), isbn, pageNumber);
+    }
+
+    @Override
+    public Book[] booksWithUserBookmarks() throws SOAPException {
+        return libraryService.booksWithUserBookmarks(getLogin()).toArray(Book[]::new);
+    }
+
+    @Override
+    public void save() {
+        libraryService.requestSerializeData();
+    }
+
+    @PreDestroy
+    private void destroy() {
+        libraryService.closeSourceService();
+    }
+
+    public String getLogin() throws SOAPException {
```

Comment:
move to AuthorizationUtil class, and make static 
```suggestion
    public static String getLoginFromRequest(WebServiceContext context) throws SOAPException {
```